packages/engine/src/datastore/batch/agent.rs

```diff
@@ -413,7 +413,7 @@ impl Batch {
         self.str_iter(column_name)
     }
 
-    pub fn get_agent_name(&self) -> Result<Vec<Option<Cow<str>>>> {
+    pub fn get_agent_name(&self) -> Result<Vec<Option<Cow<'_, str>>>> {
```

Comment:
Do we really need these anonymous lifetimes

Reply:
> I don't quite like the random anonymous lifetimes being inserted everywhere, I don't think it improves readability in most places (the ones where it renames them are fine, e.g. `'a` to `'_`). If this is a proper convention, e.g. brought in by 2018 Rust or something I'm happy to be pointed to it and probably just accept it.

It's deprecated as of Rust 2018 to omit them:

> starting with Rust 2018, it is deprecated to leave off the lifetime parameters for non-reference-types (types other than & and &mut). Instead, where you previously wrote -> StrWrap, you should now write -> StrWrap<'_>, making clear that borrowing is occurring. -- [`'_`, the anonymous lifetime](https://yegeun542.github.io/rust-edition-guide-ko/rust-2018/ownership-and-lifetimes/the-anonymous-lifetime.html)

Reply:
Fair enough